### PR TITLE
fix: ieee-rawbib fetcher

### PR DESCRIPTION
**Pull request by stefano-tech-ai**

### CODE REVIEW

**Comment by stefanomunarini** [comment]
@ronaldtse (cc @andrew2net)
As I do not have access to this repository secrets, can you please check (or point me to someone I can ask to) if amongst the secrets of this repository exists a `IETF_BIB_BOT_PAT` token? And, in particular, if this token is valid to clone the private repository https://github.com/ietf-ribose/ieee-rawbib ?

As can be seen in the GHA logs for this PR, the token currently used in the `Run bundle exec ruby crawler.rb` step seems like to not have enough permissions to clone the above mentioned repository.

**Comment by stefanomunarini** [comment]
#20

File: crawler.rb
Comment (CAMOBAP) [inline crawler.rb:10]: I'm not sure but probably `oauth2` has to be replaced with `x-access-token` i.e. 

```suggestion
system("git clone https://x-access-token:#{token}@github.com/ietf-ribose/ieee-rawbib.git ieee-rawbib")
```

**Comment by stefanomunarini** [comment]
I have added a "test" step to check if token `secrets.IETF_BIB_BOT_PAT` is present among the repository secrets (which replicate previous implementation behaviour, see https://github.com/ietf-tools/relaton-data-ieee/pull/24/files#diff-b2195247a7dfa266381b77e7cbef41ca7cbf143fd06128db55a71de7043507d0R13-R22).

It seems like this secret has been removed from this project. @CAMOBAP we might not need to use username:password (as I was suggesting [here](https://github.com/ietf-tools/relaton-data-ieee/issues/20#issuecomment-1757958070)), but simply provide this secret. Alex, could you point me at someone who has access to this repository settings?

**Comment by CAMOBAP** [comment]
@stefanomunarini IDK, @ronaldtse maybe you can suggest a right person

**Comment by stefanomunarini** [comment]
Hi @kesara , 

> It seems like this secret has been removed from this project.
> 
> I added a "test" step to check if `secrets.IETF_BIB_BOT_PAT` is present among the repository secrets (which replicate previous implementation behaviour, see https://github.com/ietf-tools/relaton-data-ieee/pull/24/files#diff-b2195247a7dfa266381b77e7cbef41ca7cbf143fd06128db55a71de7043507d0R13-R22).
> 
> @CAMOBAP we might not need to use username:password (as I was suggesting [here](https://github.com/ietf-tools/relaton-data-ieee/issues/20#issuecomment-1757958070)), but simply provide this secret and clone the repository using the canonical `system("git clone https://oauth2:#{token}@github.com/ietf-ribose/ieee-rawbib.git ieee-rawbib")`.
> 
> Alex, could you point me at someone who has access to this repository settings?

@kesara as outlined above, this repository is missing a secret, AKA `secrets.IETF_BIB_BOT_PAT`. Could you please check the settings or point me at someone that has access to this repo settings? Thanks

**Comment by kesara** [comment]
> Hi @kesara ,
> 
> > It seems like this secret has been removed from this project.
> > I added a "test" step to check if `secrets.IETF_BIB_BOT_PAT` is present among the repository secrets (which replicate previous implementation behaviour, see https://github.com/ietf-tools/relaton-data-ieee/pull/24/files#diff-b2195247a7dfa266381b77e7cbef41ca7cbf143fd06128db55a71de7043507d0R13-R22).
> > @CAMOBAP we might not need to use username:password (as I was suggesting [here](https://github.com/ietf-tools/relaton-data-ieee/issues/20#issuecomment-1757958070)), but simply provide this secret and clone the repository using the canonical `system("git clone https://oauth2:#{token}@github.com/ietf-ribose/ieee-rawbib.git ieee-rawbib")`.
> > Alex, could you point me at someone who has access to this repository settings?
> 
> @kesara as outlined above, this repository is missing a secret, AKA `secrets.IETF_BIB_BOT_PAT`. Could you please check the settings or point me at someone that has access to this repo settings, to verify that this secret is set? And, if not, help providing it? Thanks

This is set. It's a personal access token for a bot account that has access to a private repository. Earlier versions of the GHA used this to access the ieee-rawbib repository.
I guess the best way to test this is just mock it with a different private repository.
See https://github.com/ietf-tools/relaton-data-ieee/blob/5a30bfed1905c73ddcf61136f6413bac6d904d8f/.github/workflows/generate.yml

**Comment by kesara** [comment]
The secrets are not passed to workflows that are triggered by a pull request from a fork. So the PR tests will continue to fail.

**Comment by stefanomunarini** [comment]
> The secrets are not passed to workflows that are triggered by a pull request from a fork. So the PR tests will continue to fail.

I see, this was the issue then! 

Would it be possible to enable passing secrets to workflows that are triggered by a PR? This is only to test the implementation before merging the PR, it could be disabled right after testing is completed.

 @kesara

**Comment by ronaldtse** [comment]
@stefanomunarini I don't think GitHub supports passing secrets to forks due to security concerns, for example, if a public user can create a PR, that means they can send the secret elsewhere eg via a remote call.

**Comment by stefanomunarini** [comment]
> @stefanomunarini I don't think GitHub supports passing secrets to forks due to security concerns, for example, if a public user can create a PR, that means they can send the secret elsewhere eg via a remote call.

Right, makes total sense. I wonder if the only way to test this implementation is merging it? Is there no other way otherwise?

**Comment by ronaldtse** [comment]
I don't think there's a common alternative. @CAMOBAP can you please help check, thanks!

**Comment by kesara** [comment]
When this PR is ready, I'm happy to merge and I can revert the change if it doesn't work.

**Comment by stefanomunarini** [comment]
Hi @kesara , we are waiting for final review from @CAMOBAP . After he's done with it, this PR can be merged. Thanks

**Comment by CAMOBAP** [comment]
> Hi @kesara , we are waiting for final review from @CAMOBAP . After he's done with it, this PR can be merged. Thanks

@stefanomunarini if PR is ready for review please press "Ready for review" because usually "draft" PRs mean that the work is in progress and it doesn't ready for review


BTW there is no objections from my side, once workflow actually works

**Comment by stefanomunarini** [comment]
> > Hi @kesara , we are waiting for final review from @CAMOBAP . After he's done with it, this PR can be merged. Thanks
> 
> @stefanomunarini if PR is ready for review please press "Ready for review" because usually "draft" PRs mean that the work is in progress and it doesn't ready for review

You're right, that was an oversight from my side. Sorry about that

@kesara feel free to merge the PR, thanks

**Comment by CAMOBAP** [comment]
@stefanomunarini should we rerun the CI because last message was "fatal: Authentication failed for 'https://github.com/ietf-ribose/ieee-rawbib.git/'"

File: crawler.rb
Comment (kesara) [inline crawler.rb:10]: This should be `ietf-tools/ieee-rawbib`.

**Review by kesara — CHANGES_REQUESTED** [review]

File: crawler.rb
Comment (stefanomunarini) [reply crawler.rb]: Updated @kesara

**Review by kesara — APPROVED** [review]

**Comment by kesara** [comment]
GHA is running here: https://github.com/ietf-tools/relaton-data-ieee/actions/runs/6754213519/job/18361482188
There are errors but GHA is not complaining yet.

**Comment by kesara** [comment]
> GHA is running here: https://github.com/ietf-tools/relaton-data-ieee/actions/runs/6754213519/job/18361482188 There are errors but GHA is not complaining yet.

And success. Can someone confirm the errors in `Run bundle exec ruby crawler.rb  ***` are expected?

**Comment by stefanomunarini** [comment]
@CAMOBAP @andrew2net can you have a look at the comment above? Is this expected or is there anything that needs to be fixed?

**Comment by andrew2net** [comment]
> File: ieee-rawbib/updates.20230831/IEEEUpdates_IEEEstd/week27b/3185.zip

@CAMOBAP The last files from 2023 caused the error because address country not found in them. But I'm unable to get and explore these files from the ietf-tools/ieee-rawbib repo. Is there any way to get one of the files using GHA?

**Comment by CAMOBAP** [comment]
@andrew2net I think the right way will be to ask @stefanomunarini @kesara to provide required files directly (because they are in the private repo)

@stefanomunarini @kesara is it possible?

**Comment by kesara** [comment]
> @andrew2net I think the right way will be to ask @stefanomunarini @kesara to provide required files directly (because they are in the private repo)
> 
> @stefanomunarini @kesara is it possible?

I just emailed the requested file.

**Comment by andrew2net** [comment]
> File: ieee-rawbib/updates.20230831/IEEEUpdates_IEEEstd/week27b/3185.zip

The file contains an address without country:
```xml
    ...
    <address>
        <city>New York</city>
    </address>
    ...
```
@kesara @ronaldtse For the NY city we can add the USA country. All the other addresses also have US cities. It seems we can use USA as a default country, right?

**Pull request merged**
